identity/handler.go

```diff
@@ -44,6 +48,90 @@ func (h *Handler) RegisterAdminRoutes(admin *x.RouterAdmin) {
 
 	admin.POST(RouteBase, h.create)
 	admin.PUT(RouteBase+"/:id", h.update)
+
+	admin.GET(RouteMethod, h.method)
+}
+
+type methodResponse struct {
+	Method   string `json:"method"`
+	Provider string `json:"provider,omitempty"`
+}
+
+// swagger:route GET /method admin getMethod
+//
+// Get login method for a previously registered email address or username
+//
+//     Produces:
+//     - application/json
+//
+//     Schemes: http, https
+//
+//     Responses:
+//       200: loginMethod
+//       500: genericError
+func (h *Handler) method(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+
+	// if the credentials can be looked up directly by email address then they're type password
+	_, _, err := h.r.PrivilegedIdentityPool().FindByCredentialsIdentifier(r.Context(), CredentialsTypePassword, ps.ByName("id"))
```

Comment:
If I understand correctly you want to check if a user is signed up using the provided credentials, correct? In that case it would probably make sense to have a `POST` request which accepts a payload like

```
{
  "identifier": "...",
  "method": "password" # or "oidc"
}
```

and returns a `true` or `false` if it was found or not found? I think this would make the logic below much easier!

If the `method` is omitted - which means we want to discover the method - we could still implement the search algorithm you have written and respond with:

```
{
  "found": true,
  "method": "oidc",
  "oidc_providers": ["google"] // if method = oidc
}
```

The method should probably be named something like `POST /credentials/known`.

What do you think?

Reply:
Sure, that seems fine, but how about this for the response?
```
{
    "found": true,
    "methods": [
        {
            "method": "password"
        },
        {
            "method": "oidc",
            "provider": "google"
        },
        {
            "method": "oidc",
            "provider": "microsoft"
        } // ...
    ]
}
```

because there could be multiple methods to login as well as multiple OIDC providers.

Reply:
Big brain! 🧠 :) That makes total sense